### ISSUE TITLE
FormViewController is incompatible with safe area when handling keyboard events

### ISSUE THREAD

**Issue by yunhao** [xmartlabs/Eureka#1959]
- Eureka 5.1.0
- Xcode 13.3
- iOS 13.3

The example project is not updated with safe area. The incompability not only makes the UI seems strange, but also conceal a bug in FormViewController when handling keyboard events. 

<img width="33.3%" alt="Simulator Screen Shot - iPhone 11 Pro - 2019-12-25 at 14 27 13" src="https://user-images.githubusercontent.com/10205466/71435136-a93dea80-2722-11ea-86a5-7a8274aa9b0e.png">

When safe area layout guide is enabled, `tableView` should fullfill it's parrent view. `tableView` will adjust it's contentInset automatically according to safe area. 

There is a new API after iOS 11:
> **adjustedContentInset**
The insets derived from the content insets and the safe area of the scroll view.

<img width="100" alt="Screen Shot 2019-12-25 at 14 38 08" src="https://user-images.githubusercontent.com/10205466/71435491-2f0e6580-2724-11ea-92eb-357ba7edecc3.png">

And then we will find an issue when tapping a TextRow. There are some extra spacing between keyboard and active row.

<img width="33.3%" src="https://user-images.githubusercontent.com/10205466/71437320-1dc95700-272c-11ea-8d04-a17a8caa467c.gif">

These codes solve the trouble:

```swift
open class FormViewController: UIViewController, FormViewControllerProtocol, FormDelegate {
    
    @objc open func keyboardWillShow(_ notification: Notification) {
       // ...
        var newBottomInset = table.frame.origin.y + table.frame.size.height - keyBoardFrame.origin.y + rowKeyboardSpacing
        if #available(iOSApplicationExtension 11.0, *) {
            newBottomInset = newBottomInset - view.safeAreaInsets.bottom
        }
        // ...
    }
}
```

**Comment by yunhao** [comment]
There is an hacking without editing any source code.

```swift
class MyFormViewController: FormViewController {

    override func viewSafeAreaInsetsDidChange() {
        super.viewSafeAreaInsetsDidChange()
        rowKeyboardSpacing = -view.safeAreaInsets.bottom
    }

}
```

**Comment by yunhao** [comment]
#1960

**Comment by yunhao** [comment]
I create a pull request #1960

**Comment by mats-claassen** [comment]
Fixed in #1960
